governance/src/autotask.js

```diff
@@ -0,0 +1,311 @@
+        supportEmoji = checkMarkEmoji;
+      }
+
+      if (votes.length > 18) {
+        votes = votes.slice(0, votes.length - 18);
+      } else {
+        votes = '0';
+      }
+      votes = internationalNumberFormat.format(votes);
+
+      proposalName = await getProposalTitle(id);
+      if (displayName !== '') {
+        message = `**Vote** ${proposalName} ${supportEmoji} ${votes} by ${displayName} ${etherscanLink}`;
+      } else {
+        message = `**Vote** ${proposalName} ${supportEmoji} ${votes} by ${voter.slice(0, 6)} ${etherscanLink}`;
+      }
+
+      if (reason !== '') {
+        message += `\n\`\`\`${reason}\`\`\``;
+      }
+      break;
+    case 'AE-COMP-GOVERNANCE-PROPOSAL-CANCELED':
+      ({ id } = metadata);
+      proposalName = await getProposalTitle(id);
+      message = `**Canceled Proposa**l ${proposalName} ${noEntryEmoji}`;
+      break;
+    case 'AE-COMP-GOVERNANCE-PROPOSAL-EXECUTED':
+      ({ id } = metadata);
+      proposalName = await getProposalTitle(id);
+      message = `**Executed Proposal** ${proposalName} ${checkMarkEmoji}`;
+      break;
+    case 'AE-COMP-GOVERNANCE-PROPOSAL-QUEUED':
+      ({ eta, id } = metadata);
+      proposalName = await getProposalTitle(id);
+      // message = `Queued Proposal ${proposalName} ${checkMarkEmoji}
+      // available to execute after ${eta} days`;
+      message = `**Queued Proposal** ${proposalName} ${checkMarkEmoji} available to execute at timestamp ${eta}`;
+      break;
+    default:
+      return undefined;
+  }
+
+  return message;
+}
+
+// eslint-disable-next-line func-names
+exports.handler = async function (autotaskEvent) {
+  // ensure that the autotaskEvent Object exists
+  if (autotaskEvent === undefined) {
+    return {};
+  }
+  console.log('autotaskEvent');
+  console.log(JSON.stringify(autotaskEvent, null, 2));
+
+  const { secrets } = autotaskEvent;
+  if (secrets === undefined) {
+    return {};
+  }
+
+  // ensure that there is a DiscordUrl secret
+  const { TestingDiscordUrl: discordUrl } = secrets;
+  if (discordUrl === undefined) {
+    return {};
+  }
+
+  // ensure that the request key exists within the autotaskEvent Object
+  const { request } = autotaskEvent;
+  if (request === undefined) {
+    return {};
+  }
+
+  // ensure that the body key exists within the request Object
+  const { body } = request;
+  if (body === undefined) {
+    return {};
+  }
+
+  // ensure that the alert key exists within the body Object
+  const { alert } = body;
+  if (alert === undefined) {
+    return {};
+  }
+
+  // extract the transaction hash and agent ID from the alert Object
+  const {
+    hash,
+    source: {
+      transactionHash,
+      agent: {
+        id: agentId,
+      },
+    },
+  } = alert;
+
+  // retrieve the metadata from the Forta public API
+  let alerts = await getFortaAlerts(agentId, transactionHash);
+  alerts = alerts.filter((alertObject) => alertObject.hash === hash);
```

Comment:
```suggestion
```
I'm not seeing `hash` inside of the `alertObject` so this line filters out all results, making `alerts` empty. Producing zero discord messages. Commenting out or removing this line, allows it to work as intended.

Reply:
Block 14769156 for testing

Reply:
thanks changed!